### PR TITLE
allow updates to 'noauth' state

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-3846

needed for:
https://github.com/mendersoftware/deviceauth/pull/378
https://github.com/mendersoftware/integration/pull/978

Pipeline: https://gitlab.com/Northern.tech/Mender/mender-qa/-/jobs/671855381
(only 'open' tests will pass - we'll need an automerge to `-enterprise` as well)

### CODE REVIEW

File: api/http/api_inventory_test.go [hunk 1]
```diff
@@ -2315,15 +2335,15 @@ func TestApiInventoryInternalDevicesStatus(t *testing.T) {
...
+			case "accepted", "preauthorized", "pending", "noauth":
 				// Update statuses
 				inv.On("UpsertDevicesAttributes",
 					ctx,
 					tc.inputDeviceIDs,
 					deviceAttributes,
 				).Return(tc.UpdateResult, tc.inventoryErr)
 
-			case "decommissioned", "rejected":
+			case "decommissioned":
```
Comment (tranchitella) [inline api/http/api_inventory_test.go:2346]: Looks fine, just one question here: why are we not removing anymore the device on rejected?

Comment (mchalski) [reply api/http/api_inventory_test.go]: so, there are 2 things:
- we never did the removal, the test was wrong simply; if you look at the code, we delete only on decommissioned, so I just removed a mistake in test
- the recent discussion on this issue confirmed that this is what we want (keep devices around even if they're rejected, have no authsets, etc.). anyway, it was reiterated by Eystein once again

Comment (tranchitella) [reply api/http/api_inventory_test.go]: Thanks for the clarification!

Comment (mchalski) [reply api/http/api_inventory_test.go]: ...sorry, the discussion is under the original bug report:
https://tracker.mender.io/browse/MEN-2558

exchange starts here:
https://tracker.mender.io/browse/MEN-2558?focusedCommentId=108634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-108634

**Comment by merlin-northern** [comment]
the test backend failure is due to:
`/builds/Northern.tech/Mender/mender-qa/scripts/github_pull_request_status: eval: line 1: syntax error: unexpected end of file`

**Pull request merged**